osu.Game.Rulesets.Osu.Tests/TestCaseSliderInput.cs

```diff
@@ -0,0 +1,233 @@
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 2");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        /// <summary>
+        ///     Hitting a slider head, pressing a new key after the initial hit, then letting go of the new key should result
+        ///     in continue tracking,
+        ///     At 250ms intervals:
+        ///     Frame 1: Left Click
+        ///     Frame 2: Left & Right Click
+        ///     Frame 3: Left Click
+        ///     A passing test case will have the cursor continue to track after frame 3.
+        /// </summary>
+        [Test]
+        public void TestTrackingRetentionLeftRightLeft()
+        {
+            AddStep("Tracking retention test", () =>
+            {
+                var actions = new Dictionary<List<OsuAction>, double>
+                {
+                    { new List<OsuAction> { OsuAction.LeftButton }, 1500 },
+                    { new List<OsuAction> { OsuAction.LeftButton, OsuAction.RightButton }, 2500 },
+                    { new List<OsuAction> { OsuAction.LeftButton }, 3500 }
+                };
+
+                performStaticInputTest(actions);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 3");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        /// <summary>
+        ///     Pressing a key before a slider, pressing the other key on the slider head, then releasing the former pressed key
+        ///     should result in continued tracking.
+        ///     At 250ms intervals:
+        ///     Frame 1 (prior to slider):      Left Click
+        ///     Frame 2 (on slider head):       Left & Right Click
+        ///     Frame 3 (tracking slider body): Right Click
+        ///     A passing test case will have the cursor continue to track after frame 3.
+        /// </summary>
+        [Test]
+        public void TestTrackingLeftBeforeSliderToRight()
+        {
+            AddStep("Tracking retention test", () =>
+            {
+                var actions = new Dictionary<List<OsuAction>, double>
+                {
+                    { new List<OsuAction> { OsuAction.LeftButton }, 250 },
+                    { new List<OsuAction> { OsuAction.LeftButton, OsuAction.RightButton }, 1500 },
+                    { new List<OsuAction> { OsuAction.RightButton }, 2500 }
+                };
+
+                performStaticInputTest(actions);
+            });
+
+            AddUntilStep(() => allJudgedFired, "Wait for test 4");
+            AddAssert("Tracking retained", assertGreatJudge);
+        }
+
+        private bool assertMehJudge()
+        {
+            return lastResult.Type == HitResult.Meh;
+        }
+
+        private bool assertGreatJudge()
+        {
+            return lastResult.Type == HitResult.Great;
+        }
+
+        private void performStaticInputTest(Dictionary<List<OsuAction>, double> actionsOnSlider)
```

Comment:
why not just make the slider position 0 and use replayframes the whole way down?